components/gitpod-cli/cmd/tasks.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+	"google.golang.org/grpc"
+)
+
+// tasksCmd represents the tasks command
+var tasksCmd = &cobra.Command{
+	Use:   "tasks",
+	Short: "Interact with workspace tasks",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			cmd.Help()
+		}
+	},
+}
+
+var attachTaskCmdOpts struct {
+	Interactive bool
+	ForceResize bool
+}
+
+// TODO(andreafalzetti): refactor tasks.go to only keep the tasks cmd initialisation, moving the subcommands in their own file
+
+// This func was copied from https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/cmd/terminal.go#L29
+// TODO(andreafalzetti): move it somewhere else so that it's reusable from other cmds
+func dialSupervisor() *grpc.ClientConn {
+	// TODO(andreafalzetti): figure out how to import peroperly supervisor.GetConfig()
```

Comment:
Internal code of supervisor cannot be accessed in CLI, only via API.

You can see how other CLI commands connect to supervisor: https://github.com/andreafalzetti/gitpod/blob/347ad95d80f79d2e831ae9fdd3753088ee874433/components/gitpod-cli/cmd/env.go#L80-L84

Reply:
thanks, I've changed it here: https://github.com/gitpod-io/gitpod/pull/7601/commits/fadc75c03c4faaf98333e3e6a1d7dbb4ac7dd696